consumer/pubsub/src/main/java/com/spotify/heroic/consumer/pubsub/Receiver.java

```diff
@@ -43,14 +57,24 @@ public void receiveMessage(PubsubMessage message, AckReplyConsumer replyConsumer
         final ByteString data = message.getData();
         final String messageId = message.getMessageId();
         log.debug("Received ID:{} with content: {}", messageId, data.toStringUtf8());
+        final byte[] bytes = data.toByteArray();
 
         // process the data
         try {
-            consumer.consume(data.toByteArray());
-        } catch (ConsumerSchemaException e) {
+            consumer.consume(bytes);
+            reporter.reportMessageSize(bytes.length);
+            consumed.increment();
+            replyConsumer.ack();
+        } catch (ConsumerSchemaValidationException e) {
+            reporter.reportConsumerSchemaError();
             log.error("ID:{} - {}", messageId, e.getMessage(), e);
-        } finally {
+
+            // The message will never be processable, ack it to make it go away
             replyConsumer.ack();
+        } catch (Exception e) {
```

Comment:
for my understanding - removing the ack from the finally block will put the message back onto the queue to be delivered at another time, right?